addons/mail/static/src/models/attachment_list/attachment_list.js

```diff
@@ -43,11 +50,17 @@ registerModel({
         _computeImageAttachments() {
             return replace(this.attachments.filter(attachment => attachment.isImage));
         },
+        /**
+         * @returns {FieldCommand}
+         */
+        _computeLinkPreviewAttachments() {
+            return replace(this.attachments.filter(attachment => attachment.isLinkPreview));
+        },
         /**
          * @returns {Attachment[]}
          */
         _computeNonImageAttachments() {
-            return replace(this.attachments.filter(attachment => !attachment.isImage));
+            return replace(this.attachments.filter(attachment => !attachment.isImage && !attachment.isLinkPreview));
```

Comment:
This changes semantically what this field contains, so the name becomes a lie (or very misleading) if kept unchanged.

Before changes in this code, attachment list show 2 partition of attachments: images and non-images. Now you introduce a 3rd category. I'm not sure they are partitions (could have image and link preview of same attachment?), but card visual excludes attachment that are links.

Check where this field is used. If there is actually no need for "non-image attachments", then you can rename this field to match what it contains: non-image and non-link-previewable attachments.

Reply:
This should be `cardAttachment` because it regroup every attachment that will be showed as a card.

Reply:
Yeah, basically.

It's annoying to see `imageAttachment =/= attachmentImage` and `cardAttachment =/= attachmentCard`... it's confusing because prefixes "image"/"card" are adjectives on "attachment" while suffixes of "attachment" here are part of the overall concepts `AttachmentImage` and `AttachmentCard`.

But that's another problem that I'm well aware for quite some time and it's a low-priority issue for the moment.

Reply:
I agree, ideally, we should only have 1 field to handle this...